src/core/message-parser.ts

```diff
@@ -62,48 +49,32 @@ export function parseMessage(roomId: string | undefined, inboundMessage: Ably.In
     throw new Ably.ErrorInfo(`received incoming message without serial`, 50000, 500);
   }
 
-  const newMessage: ChatMessageFields = {
-    serial: message.serial,
-    clientId: message.clientId,
-    roomId,
-    text: message.data.text,
-    metadata: message.data.metadata ?? {},
-    headers: message.extras.headers ?? {},
-    createdAt: new Date(message.createdAt),
-    latestAction: message.action as ChatMessageActions,
-    latestActionSerial: message.version ?? message.serial,
-    updatedAt: message.timestamp ? new Date(message.timestamp) : undefined,
-    deletedAt: message.timestamp ? new Date(message.timestamp) : undefined,
-    operation: message.operation as MessageActionDetails,
-  };
+  if (!message.version) {
+    throw new Ably.ErrorInfo(`received incoming message without version`, 50000, 500);
+  }
 
   switch (message.action) {
-    case ChatMessageActions.MessageCreate: {
-      break;
-    }
+    case ChatMessageActions.MessageCreate:
     case ChatMessageActions.MessageUpdate:
     case ChatMessageActions.MessageDelete: {
-      if (!message.version) {
-        throw new Ably.ErrorInfo(`received incoming ${message.action} without version`, 50000, 500);
-      }
       break;
     }
     default: {
       throw new Ably.ErrorInfo(`received incoming message with unhandled action; ${message.action}`, 50000, 500);
     }
   }
+
   return new DefaultMessage(
-    newMessage.serial,
-    newMessage.clientId,
-    newMessage.roomId,
-    newMessage.text,
-    newMessage.metadata,
-    newMessage.headers,
-    newMessage.createdAt,
-    newMessage.latestAction,
-    newMessage.latestActionSerial,
-    newMessage.latestAction === ChatMessageActions.MessageDelete ? newMessage.deletedAt : undefined,
-    newMessage.latestAction === ChatMessageActions.MessageUpdate ? newMessage.updatedAt : undefined,
-    newMessage.operation,
+    message.serial,
+    message.clientId,
+    roomId,
+    message.data.text,
+    message.data.metadata ?? {},
+    message.extras.headers ?? {},
+    message.action as ChatMessageActions,
+    message.version,
+    new Date(message.createdAt),
```

Comment:
Do we still need a check above to ensure createdAt and timestamp are present?

Reply:
We might do a cleanup of this function in the future

Reply:
Turns out we don't need the MessagePayload interface anymore so I've removed that